lms/djangoapps/instructor/views/api.py

```diff
@@ -726,6 +728,343 @@ def extract_user_info(user):
     return JsonResponse(response_payload)
 
 
+def build_course_tree(course):
+    """
+    Recursively builds a course tree given the root node to a course
+    :param course tree root:
+    :return list of dictionaries that represent the tree:
+    """
+    course_tree = []
+    idx = 0
+    for child in course.get_children():
+        if not child.hide_from_toc:
+            using = course.children[idx]
+            course_tree.append({'display_name': child.display_name_with_default,
+                               'block_id': using.block_id,
+                               'block_type': using.block_type,
+                               'sub': build_course_tree(child)})
+        idx += 1
+    return course_tree
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+@require_query_params(rolename="'instructor', 'staff', or 'beta'")
+def list_course_tree(request, course_id):
+    """
+    Returns a tree representing the substructures in a course
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_with_access(request.user, 'instructor', course_id, depth=None)
+    course_tree = build_course_tree(course)
+    response_payload = {
+        'course_id': course_id.to_deprecated_string(),
```

Comment:
As above, why are we using the deprecated form instead of `unicode(course_id)` or whatever the newest format is?